src/features/shared-link-settings-modal/AllowDownloadSection.js

```diff
@@ -5,16 +5,19 @@ import { FormattedMessage } from 'react-intl';
 import Checkbox from '../../components/checkbox';
 import TextInputWithCopyButton from '../../components/text-input-with-copy-button';
 import Fieldset from '../../components/fieldset';
+import Tooltip from '../../components/tooltip';
 
 import messages from './messages';
 
 const AllowDownloadSection = ({
     canChangeDownload,
+    classification,
     directLink,
     directLinkInputProps = {},
     downloadCheckboxProps = {},
     isDirectLinkAvailable,
     isDirectLinkUnavailableDueToDownloadSettings,
+    isDirectLinkUnavailableDueToAccessPolicy = false,
```

Comment:
@priyajeet is this component suppose to be so specific? I am curious if this could just be `isDirectLinkDisabled` and then have a prop of `reason`

Reply:
Its an older component :/ Could use some refactoring, also see all those proptypes at the bottom. But making all those changes is going way beyond the scope